### DISCUSSION TITLE
Possible Memory Leak?

### DISCUSSION THREAD

**Discussion by surjithctly** [umami-software/umami#1167] [General]
I'm using PostgresDB and it seems that its using 100% resources all the time. I restarted it once and it went to normal, but then after few days it became full. See screenshot. Is this might have happened because of memory leaks? 

Let me know what info you need to debug, like some SQL commands, I can do that. 

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/1884712/168037722-b0f86b9e-d105-4bb3-9653-2f7544095f44.png">

See how memory & network is using the max available. My websites are not went viral, I have added 5 sites, and gets a maximum of 2k visits per day.

**Comment by ruibaby** [comment]
same here https://github.com/mikecao/umami/issues/1134

**Comment by dipakparmar** [comment]
@surjithctly How many connections you're having on postgres from umami?

**Comment by surjithctly** [comment]
@dipakparmar How can I check them? Only one umami site is connected.

**Comment by dipakparmar** [comment]
> @dipakparmar How can I check them? Only one umami site is connected.

@surjithctly you could check by running this query or if you have pgadmin installed it will show you:

```sql
select pid as process_id, 
       usename as username, 
       datname as database_name, 
       client_addr as client_address, 
       application_name,
       backend_start,
       state,
       state_change
from pg_stat_activity;
```

**Comment by surjithctly** [comment]
Here are the results: @dipakparmar 

<img width="1076" alt="image" src="https://user-images.githubusercontent.com/1884712/169974455-f5d1370e-ea69-4228-97c0-e8e5b7102aaa.png">
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/1884712/169974498-94e3d4d4-1c21-48d4-bc6f-250dc5eef9f5.png">

**Reply by dipakparmar** [reply]
seems normal number connections! is the usage that you showed on earlier screenshot is during peak time or regular?

**Reply by surjithctly** [reply]
It's regular.  Drops to normal after a restart but picks up to max in a day or two.

**Reply by surjithctly** [reply]
I'm using railway.app

**Reply by dipakparmar** [reply]
In that case, check with Railway support. You could also try running the DB on docker local and test with simulated traffic.

**Reply by surjithctly** [reply]
okay. will check with them. Not a docker expert :)

**Comment by surjithctly** [comment]
If anyone using Railway (recommended by Umami) as the database, please share your metrics screen! I wanted to see if its just me. 

Coz its not solved yet.

**Reply by taylor-stein** [reply]
Unsure if it's related, but I'm seeing memory constantly rise until the server crashes, or until the deployment is automatically restarted. Followed the docs for configuring Umami with Railway. You can see the crash at around 6 AM: 

![image](https://user-images.githubusercontent.com/59148531/182003368-bfc98ba7-db5e-428b-8a01-1b6c3e203575.png)

**Reply by rylax** [reply]
Any update on this? Have you figured out the root cause? I am experiencing the exact same.

**Reply by surjithctly** [reply]
I had the issue while importing data from my old sql server. I created a new server without backup or import. just fresh. Then it worked as expected

**Reply by rylax** [reply]
Hm, I see. I started everything fresh and this keeps on happening unfortunately. The only logs I get are ```TypeError: Cannot destructure property 'session_id' of 'session' as it is null``` but not sure they are even related with the leak.

**Reply by dipakparmar** [reply]
@rylax try running failed migrations manually if you have any failed. I had similar problems after the upgrade and had to run migrations manually, unfortunately automatically migration did not work properly.
